### DISCUSSION TITLE
Need a guide on how to encrypt emails and also how the user wlll decrypt emails

### DISCUSSION THREAD

**Discussion by aravind-25-gopinath** [openpgpjs/openpgpjs#1361] [Q&A]
Hi, I need to send an email (currently using nodejs and nodemailer to send) that needs to be encryped. I found openpgp is an encryption library but have not seen any guide on how to implement it for an email. Could I get some assistance please?

**Comment by larabr** [comment]
It depends on what you need exactly. The simplest thing to implement would be to get the armored encrypted message, add it to the email body, and send the email. The recipient can copy-paste the armored message and use any OpenPGP library/software to decrypt it. This approach is referred to as "Inline PGP".

If you want the email client to be aware that the message contains OpenPGP-encrypted data, you need to generate emails that follow the PGP/MIME format (https://datatracker.ietf.org/doc/html/rfc3156).

Here you have a high-level overview of the two approaches: https://protonmail.com/support/knowledge-base/pgp-mime-pgp-inline/

**Reply by aravind-25-gopinath** [reply]
What I was thinking was that I send an encrypted email to the user, and once the user receives it they can use a private key to decrypt the message. Is this possible?

**Reply by larabr** [reply]
Yes, both of the approaches I mentioned require the user to have the private key to decrypt :)
The main issue in integrating OpenPGP with email is about how you want to encode the encrypted data basically, that's where PGP/Inline vs PGP/MIME come into play.

**Reply by aravind-25-gopinath** [reply]
Are there any code examples of this I can take a look at? I am new to encryption

**Reply by larabr** [reply]
You can encrypt and decrypt a message as shown here: https://github.com/openpgpjs/openpgpjs#encrypt-and-decrypt-string-data-with-pgp-keys.
The encrypted data you get is armored, meaning it is in string form, and you can just copy-paste that in the body of the email.

If you need to generate the private and public keys for the user, see: https://github.com/openpgpjs/openpgpjs#generate-new-key-pair

**Reply by aravind-25-gopinath** [reply]
Do you know if there is any integration with nodemailer that can be used?

**Reply by larabr** [reply]
I'm not familiar with it, I don't know if there is an integration that uses our library, but I've found this one: https://github.com/nodemailer/nodemailer-openpgp
